ndcube/ndcube.py

```diff
@@ -792,3 +801,90 @@ def plot(self, *args, **kwargs):
                 "no default plotting functionality is available.")
 
         return self.plotter.plot(*args, **kwargs)
+
+    def _new_instance_from_op(self, new_data, new_unit):
+        # This implicitly assumes that the arithmetic operation does not alter
+        # the WCS, mask, or uncertainty
+        new_cube = type(self)(new_data,
+                              unit=new_unit,
+                              wcs=self.wcs,
+                              mask=self.mask,
+                              meta=self.meta,
+                              uncertainty=self.uncertainty)
+        if self.extra_coords is not None:
+            new_cube._extra_coords = deepcopy(self.extra_coords)
+        return new_cube
+
+    def __neg__(self):
+        return self._new_instance_from_op(-self.data, self.unit)
+
+    def __pow__(self, value):
+        new_data = self.data ** value
+        new_unit = self.unit ** value
+        return self._new_instance_from_op(new_data, new_unit)
+
+    def __add__(self, value):
+        if hasattr(value, 'unit'):
+            if isinstance(value, u.Quantity):
+                # NOTE: if the cube does not have units, we cannot
+                # perform arithmetic between a unitful quantity.
+                # This forces a conversion to a dimensionless quantity
+                # so that an error is thrown if value is not dimensionless
+                cube_unit = u.Unit('') if self.unit is None else self.unit
+                new_data = self.data + value.to_value(cube_unit)
+            else:
+                # NOTE: This explicitly excludes other NDCube objects and NDData objects
+                # which could carry a different WCS than the NDCube
+                return NotImplemented
+        else:
+            new_data = self.data + value
+        return self._new_instance_from_op(new_data, self.unit)
+
+    def __radd__(self, value):
+        return self.__add__(value)
+
+    def __sub__(self, value):
+        return self.__add__(-value)
+
+    def __rsub__(self, value):
+        return self.__neg__().__add__(value)
+
+    def __mul__(self, value):
+        if hasattr(value, 'unit'):
+            if isinstance(value, u.Quantity):
+                # NOTE: if the cube does not have units, set the unit
+                # to dimensionless such that we can perform arithmetic
+                # between the two.
+                cube_unit = u.Unit('') if self.unit is None else self.unit
+                new_data = self.data * value.to_value()
+                new_unit = cube_unit * value.unit
```

Comment:
I think this highlights that we probably want an `NDCube.to` method as users will likely want normalize data e.g. with units of `u.m / u.cm`.  Further use cases include converting between spectral units.  However, this is probably beyond the scope of this PR.

Reply:
I'm in favor of this, but I agree it is out of scope here.